code/game/machinery/vending.dm

```diff
@@ -305,9 +305,14 @@
 			attack_hand(user)
 		return
 	else if(istype(W, /obj/item/weapon/coin) && premium.len > 0)
-		user.drop_item(src)
-		coin = W
-		user << "<span class='notice'>You insert [W] into [src].</span>"
+		if (isnull(coin))
+			user.drop_item()

```

Comment:
Better standard to use drop_item(src).
It's the same effect most of the time, but it makes code maintenance easier.